cmd/trust/sudi.go

```diff
@@ -3,122 +3,106 @@ package main
 import (
 	"crypto/x509"
 	"crypto/x509/pkix"
-	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
 	"time"
 
-	"github.com/apex/log"
 	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/project-machine/trust/pkg/trust"
 	"github.com/urfave/cli"
 )
 
-var genSudiCmd = cli.Command{
-	Name:      "gen-sudi",
+var sudiCmd = cli.Command{
+	Name:      "sudi",
 	Usage:     "Generate and sign sudi cert",
 	UsageText: "CACert, private-key, output-dir, product-uuid [, machine-uuid]",
-	Action:    doGenSudi,
+	Subcommands: []cli.Command{
+		cli.Command{
+			Name:      "list",
+			Action:    doListSudi,
+			Usage:     "list sudi keys",
+			ArgsUsage: "<keyset-name> <project-name>",
+		},
+		cli.Command{
+			Name:      "add",
+			Action:    doGenSudi,
+			Usage:     "add a new sudi key to project",
+			Flags: []cli.Flag{
+				cli.StringFlag {
+					Name: "uuid",
+					Usage: "specify a machine-uuid.  If unspecified, use random.",
+				},
+			},
+			ArgsUsage: "<keyset-name> <project-name> <serial-number>",
```

Comment:
are we naming a VM or giving it any kind of string or numeral representation when we create it? Maybe we have and that is what serial-number is.  Otherwise the user would have to create or makeup serial numbers for every VM and keep track of them. Just wondering.

Reply:
Ye,s that's the serial number or uuid.

As I commented above, I need to tweak that, as right now I'm creating both.  It's one or the other - a user-provided serial number, or auto-generated uuid.